Fix for CR-1214058

aktondak opened the pull request:
<!-- Thanks for sending a pull request! Please fill out below, remove sections that don't apply for your pull request.  -->
#### Problem solved by the commit
Fixes CR-1214058 
Print before fix :
```
>xrt-smi examine -e
ERROR: the required argument for option '--element' is missing

COMMAND: examine

DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
               in a text or JSON format.

USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --help             - Help to use this sub-command
  -r, --report       - The type of report to be produced. Reports currently available are:
                       Common:
                         all             - All known reports are produced
                         host            - Host information
                         platform        - Platforms flashed on the device
                       AIE:
                         aie-partitions  - AIE partition information
                         telemetry       - Telemetry data for the device
                       Alveo:
                         aie             - AIE metadata in xclbin
                         aiemem          - AIE memory tile information
                         aieshim         - AIE shim tile status
                         debug-ip-status - Status of Debug IPs present in xclbin loaded on device
                         dynamic-regions - Information about the xclbin and the compute units
                         electrical      - Electrical and power sensors present on the device
                         error           - Asyncronus Error present on the device
                         firewall        - Firewall status
                         mailbox         - Mailbox metrics of the device
                         mechanical      - Mechanical sensors on and surrounding the device
                         memory          - Memory information present on the device
                         pcie-info       - Pcie information of the device
                         qspi-status     - QSPI write protection status
                         thermal         - Thermal sensors present on the device

GLOBAL OPTIONS:
  -verbose           Turn on verbosity
  -batch             Enable batch mode (disables escape characters)
  -force             When possible, force an operation
ERROR: operation canceled
```
After Fix :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi examine -e
System Configuration
  OS Name              : Windows NT
  Release              : 26100
  Machine              : x86_64
  CPU Cores            : 16
  Memory               : 64878 MB
  Distribution         : Microsoft Windows 11 Pro
  Model                : KoratPlus-KRK
  BIOS Vendor          : AMD
  BIOS Version         : WXC4703N

XRT
  Version              : 2.18.0
  Branch               : CR-1214058
  Hash                 : 8c04d6b082ddb934953e1cf18c74ea466f9886ca
  Hash Date            : 2024-10-01 15:23:01
  NPU Driver Version   : 10.1.0.1
  NPU Firmware Version : 0.7.37.42

Device(s) Present
|BDF             ||Name  |
|----------------||------|
|[00c5:00:01.1]  ||NPU   |

```
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
https://jira.xilinx.com/browse/CR-1214058

#### How problem was solved, alternative solutions (if any) and why they were rejected
The problem was solved by providing implicit empty values to options which are ryzenAI subcmd specific. This was we only print ryzenAI specific help.
This is a similar fix as the PR : https://github.com/Xilinx/XRT/pull/8317

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Tested on strix and kracken boards 

#### Documentation impact (if any)
None

gbuildx commented:
Can one of the admins verify this patch?

AShivangi commented:
Hmm, but it doesn't shows the help menu. Also, what do you get when you pass in an option that is not supported, eg: xrt-smi examine -q

aktondak commented:
> Hmm, but it doesn't shows the help menu. Also, what do you get when you pass in an option that is not supported, eg: xrt-smi examine -q

This doesn't show an error because empty -e|--element is now considered a valid option which defaults to nothing being filtered.
-e option refers to : "Filters individual elements(s) from the report. Format: '/<key>/<key>/...'"

 xrt-smi examine -q gives correct behavior
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi examine -q
ERROR: Unrecognized arguments:
  -q


COMMAND: examine

DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
               in a text or JSON format.

USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --help             - Help to use this sub-command
  -r, --report       - The type of report to be produced. Reports currently available are:
                         aie-partitions - AIE partition information
                         all            - All known reports are produced
                         host           - Host information
                         platform       - Platforms flashed on the device
                         telemetry      - Telemetry data for the device

```

aktondak commented:
> Can we mimic what `xrt-smi examine -o` is doing?
> 
> ```
> >xrt-smi examine -o
> ERROR: Output file not specified: invalid argument
> 
> COMMAND: examine
> 
> DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
>                in a text or JSON format.
> 
> USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]
> 
> OPTIONS:
>   -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
>   -f, --format       - Report output format. Valid values are:
>                          JSON        - Latest JSON schema
>                          JSON-2020.2 - JSON 2020.2 schema
>   -o, --output       - Direct the output to the given file
>   --help             - Help to use this sub-command
>   -r, --report       - The type of report to be produced. Reports currently available are:
>                          aie-partitions - AIE partition information
>                          all            - All known reports are produced
>                          host           - Host information
>                          platform       - Platforms flashed on the device
>                          telemetry      - Telemetry data for the device
> 
> 
> GLOBAL OPTIONS:
>   --verbose          - Turn on verbosity
>   --batch            - Enable batch mode (disables escape characters)
>   --force            - When possible, force an operation
> ERROR: operation canceled
> ```

Sounds good.
I have updated the behavior. I have also updated the desciption of the PR with new behavior

AShivangi reviewed (approved):
Thanks! LGTM!
